server/middleware/isUserAdmin.js

```diff
@@ -0,0 +1,44 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import { User } from '../db/models';
+
+dotenv.config();
+
+const isUserAdmin = async (req, res, next) => {
+  const token = req.header('x-auth-token');
+  if (!token) {
+    return res.status(401).json({
+      status: 401,
+      message: 'System rejected. No access token found!',
+    });
+  }
+
+  try {
+    const { email, isAdmin } = jwt.verify(token, process.env.JWT_KEY);
+    const  user = await User.findAll({
+      where: {
+        email: email.trim(),
+      },
+    });
+    if (user.length === 0) {
+      return res.status(401).json({
+        status: 401,
+        message: 'Awww, Snap!..Such kind of access token does not match any user!'
+      });
```

Comment:
as I mentioned before please try and return errors, not messages on failing
if possible change the error/message strings you are using to be more understandable and general for this middleware might be used again 
for example

_message: 'Awww, Snap!..Such kind of access token does not match any user!_
most of the users don't even know what a token is.
from my point of view even _**error**: "access denied"_ would be better 

may the same apply to all errors/messages to be returned

Reply:
@blaise82 thank you for such correction. I am on it